design/proof-erasure-coding.md

```diff
@@ -0,0 +1,419 @@
+Storage proofs & erasure coding
+===============================
+
+Authors: Codex Team
+
+Erasure coding is used for multiple purposes in Codex:
+
+- To restore data when a host drops from the network; other hosts can restore
+  the data that the missing host was storing.
+- To speed up downloads
+- To increase the probability of detecting missing data on a host
+
+For the first two items we'll use a different erasure coding scheme than we do
+for the last. In this document we focus on the last item; an erasure coding
+scheme that makes it easier to detect missing or corrupted data on a host
+through storage proofs.
```

Comment:
I'm also not yet convinced we can say "we use different erasure coding" and we can handle them separately. In the construct I'm thinking of I'm not sure the erasure coding can be separated this way.

Reply:
> what we call a **slot**

Thanks, I've used 'slot data' in the updated version

> not yet convinced we can say "we use different erasure coding"

In this proposed design we use a different erasure coding. It would be really cool if we could use a single erasure coding scheme for all purposes, but I haven't seen a design for it yet?